packages/reputation-miner/ReputationMiner.js

```diff
@@ -603,7 +603,7 @@ class ReputationMiner {
       }
     }
     if (!entryIndex) {
-      return new Error("No valid entry for submission found");
+      throw new Error("No valid entry for submission found.");
```

Comment:
If I'm reading this right, `entryIndex` will always have a value if [this](https://github.com/JoinColony/colonyNetwork/pull/508/files#diff-9c228e1a1f7fe6683164300c770b62ffR593) condition passes even once. Therefore the only time we throw an error is when the above-mentioned condition  is never true?

Reply:
Hm, entryIndex should actually only be set after the estimate for `submitRootHash` succeeds.

Reply:
👆 I agree

Reply:
Mkay I can update. And yeah - if there's no stake, then we never even loop and so this is `undefined`. If there is only a small stake, then we may not find a valid index, and so `undefined`.

Reply:
Ok I fixed.